pycbc/events/stat.py

```diff
@@ -1716,26 +1745,128 @@ def get_statistic(stat):
         raise RuntimeError('%s is not an available detection statistic' % stat)
 
 
-def get_sngl_statistic(stat):
+def insert_statistic_option_group(parser):
+    """
+    Add ranking statistic options to the optparser object.
+
+    Adds the options used to initialize a PyCBC Stat class.
+
+    Parameters
+    -----------
+    parser : object
+        OptionParser instance.
+
+    Returns
+    --------
+    strain_opt_group : optparser.argument_group
+        The argument group that is added to the parser.
+    """
+
+    statistic_opt_group = parser.add_argument_group(
+        "Options needed to initialize a PyCBC Stat class for computing the "
+        "ranking of events from a PyCBC search."
+    )
+
+    statistic_opt_group.add_argument(
+        "--ranking-statistic",
+        choices=statistic_dict.keys(),
+        required=True,
+        help="The coinc ranking statistic to calculate"
+    )
+
+    statistic_opt_group.add_argument(
+        "--sngl-ranking",
+        choices=ranking.sngls_ranking_function_dict.keys(),
+        required=True,
+        help="The single-detector trigger ranking to use."
+    )
+
+    statistic_opt_group.add_argument(
+        "--statistic-files",
+        nargs='*',
+        action='append',
+        default=[],
+        help="Files containing ranking statistic info"
+    )
+
+    statistic_opt_group.add_argument(
+        "--statistic-keywords",
+        nargs='*',
+        default=[],
+        help="Provide additional key-word arguments to be sent to "
+             "the statistic class when it is initialized. Should "
+             "be given in format --statistic-keywords "
+             "KWARG1:VALUE1 KWARG2:VALUE2 KWARG3:VALUE3 ..."
+    )
+
+    return statistic_opt_group
+
+def parse_statistic_keywords_opt(stat_kwarg_list):
     """
-    Error-handling sugar around dict lookup for single-detector statistics
+    Parse the list of statistic keywords into an appropriate dictionary.
+
+    Take input from the input argument ["KWARG1:VALUE1", "KWARG2:VALUE2",
+    "KWARG3:VALUE3"] and convert into a dictionary.
 
     Parameters
     ----------
-    stat : string
-        Name of the single-detector statistic
+    stat_kwarg_list : list
+        Statistic keywords in list format
+
+    Returns
+    -------
+    stat_kwarg_dict : dict
+        Statistic keywords in dict format
+    """
+    stat_kwarg_dict = {}
+    for inputstr in stat_kwarg_list:
+        try:
+            key, value = inputstr.split(':')
+            stat_kwarg_dict[key] = value
+        except ValueError:
+            err_txt = "--statistic-keywords must take input in the " \
+                      "form KWARG1:VALUE1 KWARG2:VALUE2 KWARG3:VALUE3 ... " \
+                      "Received {}".format(opts.statistic_keywords)
+            raise ValueError(err_txt)
+
+    return stat_kwarg_dict
+
+def get_statistic_from_opts(opts, ifos):
```

Comment:
as above, this would be more consistent with other suggestion as get_ranking_from_opts